driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -1593,15 +1593,15 @@ private void connect() {
                 if (serverResponseVersion == attemptedVersion) {
                     // We're post CASSANDRA-11464: the server replies with the same version that the client tried (to
                     // ensure that the client can decode the response).
-                    // Retry with a lower version (if it's too low we'll eventually reach our min and fail).
+                    // Retry with a lower version (if it's too low we'll fail at the next attempt, as we only retry once).
```

Comment:
Sorry, but your suggestion makes some tests fail. The main reason is that we do not retry more than once, so if we start too high, the second attempt will still be too high and fail, whereas currently the second attempt could potentially be much lower that the initial one. 

On the light of this, I'm actually reconsidering my position: I think we should rather keep the current algorithm.